commerce-subscription-web/src/main/java/com/liferay/commerce/subscription/web/internal/display/context/CPDefinitionSubscriptionInfoDisplayContext.java

```diff
@@ -85,6 +99,48 @@ public PortletURL getPortletURL() throws PortalException {
 		return portletURL;
 	}
 
+	public boolean hasRecurringPaymentMethod() throws PortalException {
+		List<CommerceChannelRel> commerceChannelRels =
+			CommerceChannelRelLocalServiceUtil.getCommerceChannelRels(
+				CPDefinition.class.getName(), getCPDefinitionId(), -1, -1,
+				null);
+
+		for (CommerceChannelRel commerceChannelRel : commerceChannelRels) {
```

Comment:
This code is really hard to maintain. I don't like it, let's find an easier implementation

Reply:
As discussed in slack I'm not sure what other approach we can take